### PR TITLE
[4] Avoid the EOL underline of the title link

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue # .

### Summary of Changes
Breaking  lines for an anchor link causes a blank for underline 

### Testing Instructions
Make a module for atricles categoriy on your site. Hover the title and see the underline .

### Actual result BEFORE applying this Pull Request
link-eol

### Expected result AFTER applying this Pull Request
link-eol2

### Documentation Changes Required
no

### CODE REVIEW

**Comment by chmst** [comment]
Note: It is an a11y issue that links are not underlined in this module - but that is not in scope here

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 13af1a45437633079adc89f4654960463d7f4b32<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35514">issues.joomla.org/tracker/joomla-cms/35514</a>.</sub>

**Comment by brianteeman** [comment]
I wonder if we can find a better way to resolve this throughout the CMS without impacting on the readability of the code. Perhaps if the html was _compressed_ before being rendered by the browser?

**Comment by chmst** [comment]
I'll try .. we have the same issue in mod_syndication, where we have no chance with a single line. 
syndication

**Comment by brianteeman** [comment]
exactly

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 13af1a45437633079adc89f4654960463d7f4b32<br /><br />@chmst thanks for assistance :-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35514">issues.joomla.org/tracker/joomla-cms/35514</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35514">issues.joomla.org/tracker/joomla-cms/35514</a>.</sub>

**Comment by chmst** [comment]
Back to pending. Using HtmlHelper is better style.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35514">issues.joomla.org/tracker/joomla-cms/35514</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 7dad341afff8a3be022f107baa5513abcd477e32<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35514">issues.joomla.org/tracker/joomla-cms/35514</a>.</sub>

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on 7dad341afff8a3be022f107baa5513abcd477e32<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35514">issues.joomla.org/tracker/joomla-cms/35514</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35514">issues.joomla.org/tracker/joomla-cms/35514</a>.</sub>

File: modules/mod_articles_category/tmpl/default_items.php [hunk 1]
```diff
@@ -17,9 +17,15 @@
...
-		</a>
+		<?php
+			$attributes = [
+				'class' => 'mod-articles-category-title ' . $item->active
+			];
+			$link = htmlspecialchars($item->link, ENT_COMPAT, 'UTF-8', false);
+			$title = htmlspecialchars($item->title, ENT_COMPAT, 'UTF-8', false);
+
+			echo HTMLHelper::_('link', $link, $title, $attributes);
+		?>
```
Comment (Quy) [inline modules/mod_articles_category/tmpl/default_items.php]:
```suggestion
		<?php $attributes = ['class' => 'mod-articles-category-title ' . $item->active]; ?>
		<?php $link = htmlspecialchars($item->link, ENT_COMPAT, 'UTF-8', false); ?>
		<?php $title = htmlspecialchars($item->title, ENT_COMPAT, 'UTF-8', false); ?>
		<?php echo HTMLHelper::_('link', $link, $title, $attributes); ?>
```

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**